huff_parser/tests/opcodes.rs

```diff
@@ -34,3 +34,115 @@ fn not_mistaken_as_opcode() {
         assert_eq!(actual_label, TokenKind::Label(label));
     }
 }
+
+#[test]
+#[should_panic]
+fn test_invalid_push_non_literal() {
+    let source: &str = r#"
+        // Here we have a macro invocation directly in the parameter list - this should fail
+        #define macro MAIN() = takes (0) returns (0) {
+            push1 0x10
+            push32 0x108
```

Comment:
One thing I noticed- the literals aren't automatically padded to the `push<n>` size, causing unexpected behavior. For ex, this `push32` op will consume the `0x0108` and then the 30 bytes after that, where it should pad the `0x0108` to `0x0000000000000000000000000000000000000000000000000000000000000108`

Reply:
Fixed overflows and underflows